tests/snowflake/operators/test_snowflake_async.py

```diff
@@ -34,31 +40,58 @@ def context():
     yield context
 
 
+def create_context(task, dag):
+    execution_date = datetime.datetime(2022, 1, 1, 0, 0, 0)
+    dag_run = DagRun(
+        dag_id=dag.dag_id,
+        execution_date=execution_date,
+        run_id=DagRun.generate_run_id(DagRunType.MANUAL, execution_date),
+    )
+    task_instance = TaskInstance(task=task)
+    task_instance.dag_run = dag_run
+    task_instance.dag_id = dag.dag_id
+    task_instance.xcom_push = mock.Mock()
+    return {
+        "dag": dag,
+        "run_id": dag_run.run_id,
+        "task": task,
+        "ti": task_instance,
+        "task_instance": task_instance,
+    }
+
+
+@pytest.mark.parametrize("mock_sql", [TEST_SQL, [TEST_SQL]])
 @mock.patch(LONG_MOCK_PATH)
-def test_snowflake_execute_operator_async(context):
+def test_snowflake_execute_operator_async(mock_db_hook, mock_sql):
     """
     Asserts that a task is deferred and an SnowflakeTrigger will be fired
     when the SnowflakeOperatorAsync is executed.
     """
+    args = {"owner": "airflow", "start_date": datetime.datetime(2017, 1, 1)}
+    dag = DAG("test_snowflake_async_execute_complete_failure", default_args=args)
 
     operator = SnowflakeOperatorAsync(
         task_id="execute_run",
         snowflake_conn_id=CONN_ID,
-        sql=TEST_SQL,
+        dag=dag,
```

Comment:
do we need this dag param here?

Reply:
Yes, because the extract method in SnowflakeAsyncExtractor class extract the DAG details